js/content/community.js

```diff
@@ -3700,6 +3700,98 @@ let WorkshopBrowseClass = (function(){
+        HTML.beforeBegin("#es_add_tag", tag);
+    }
+
+    EditGuidePageClass.prototype.allowMultipleLanguages = function() {
+        document.getElementsByName("tags[]").forEach(tag => tag.type = "checkbox");
+    };
+
+    EditGuidePageClass.prototype.addCustomTags = function() {
+        let langSection = document.querySelector("#checkboxgroup_1");
+        if (!langSection) { return; }
+
+        Messenger.addMessageListener("addtag", function(name) {
+            addTag(name);
+        });
+        
+        HTML.afterEnd(langSection,
+            `<div class="tag_category_container" id="checkboxgroup_2">
+                <div class="tag_category_desc">${Localization.str.custom_tags}</div>
+                <div><a style="margin-top: 8px;" class="btn_blue_white_innerfade btn_small_thin" id="es_add_tag">
+                    <span>${Localization.str.add_tag}</span>
+                </a></div>
+            </div>`);
+
+        document.querySelector("#es_add_tag").addEventListener("click", function() {
+            ExtensionLayer.runInPageContext(`function() {
+                let Modal = ShowConfirmDialog("${Localization.str.custom_tags}", \`<div class="commentthread_entry_quotebox"><textarea placeholder="${Localization.str.enter_tag}" class="commentthread_textarea" id="es_tag" rows="1"></textarea></div>\`);
+                
+                let tag = "";
+                function done() {
+                    if (tag.trim().length === 0) { return; }
+                    tag = tag[0].toUpperCase() + tag.slice(1);
+                    Messenger.postMessage("addtag", tag);
+                }
+
+                $J("#es_tag").on("keyup paste input", function(e) {
+                    let code = e.keyCode || e.which;
+                    if (code == 13) {
+                        Modal.Dismiss();
+                        done();
+                        return;
+                    }
+
+                    tag = $J("#es_tag").val();
+                });
+
+                Modal.done(done);
+            }`);
+        });
+    };
+
+    EditGuidePageClass.prototype.rememberTags = function() {
+        let submitBtn = document.querySelector("[href*=SubmitGuide]");
+        if (!submitBtn) { return; }
+
+        let params = new URLSearchParams(window.location.search);
+        let id = params.get("id") || "recent";
+        let savedTags = JSON.parse(LocalStorage.get("es_guide_tags") || "{}");
+        if (!savedTags[id]) {
+            savedTags[id] = savedTags.recent || [];
+        }
+
+        for (let tag of savedTags[id]) {
+            let node = document.querySelector(`[name="tags[]"][value="${tag}"]`);
+            if (node) {
+                node.checked = true;
+            } else {
+                addTag(tag);
+            }
+        }
+
+        submitBtn.removeAttribute("href");
+        submitBtn.addEventListener("click", function() {
```

Comment:
Shouldn't you save the tags already when creating them instead of only saving when submitting?

Reply:
No, it only does that for 1 language and 1 category tag. Not all the other tags. Try yourself.